File: lectrick/tiles/light_block.py
from .one_way_base import OneWayTile


class LightBlock(OneWayTile):
    TILE_TYPE = "LIGHT BLOCK"
    SAMPLE_CHARS = ["⬛", "⯀", "◼", "▮"]

    def __init__(self, x, y, tiles):
        super().__init__(x, y, tiles)

    def start_turn(self):
        pass

    def end_turn(self):
        if self.current_input:
            amount, source_x, source_y = self.current_input
            print(chr(amount), end="", flush=True)

File: web/src/tiles/lightBlock.ts
import {OneWayBase} from './oneWayBase';

export class LightBlockTile extends OneWayBase {
    TILE_TYPE = '"LIGHT BLOCK';
    SAMPLE_CHARS = ["⬛", "⯀", "◼", "▮"];

    constructor(ctx: ExecutionContext, x: number, y: number) {
        super(ctx, x, y);
    }

    endTurn() {
        if (this.currentInput) {
            const amount = this.currentInput[0];
            this.ctx.print(String.fromCharCode(amount));
        }
    }
}